designs/2024-report-unused-inline-configs/README.md

```diff
@@ -0,0 +1,261 @@
+It assumes the `accessor-pairs` default options from [feat: add meta.defaultOptions](https://github.com/eslint/eslint/pull/17656) of:
+
+```json
+{
+  "enforceForClassMembers": true,
+  "getWithoutSet": false,
+  "setWithoutGet": true
+}
+```
+
+<table>
+  <thead>
+    <tr>
+      <th>Original Config Setting</th>
+      <th>Inline Config Settings</th>
+      <th>Report?</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td rowspan="7">
+        <code>"error"</code>
+        <br />
+        <code>2</code>
+        <br />
+        <code>["error", { "enforceForClassMembers": true }]</code>
+        <br />
+        <code>["error", { "getWithoutSet": false }]</code>
+      </td>
+      <td>
+        <code>"off"</code>
+        <br />
+        <code>0</code>
+      </td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>
+        <code>"warn"</code>
+        <br />
+        <code>1</code>
+      </td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>
+        <code>"error"</code>
+        <br />
+        <code>2</code>
+      </td>
+      <td><strong>Yes</strong></td>
+    </tr>
+    <tr>
+      <td>
+        <code>["error", { "enforceForClassMembers": false }]</code>
+        <br />
+        <code>[2, { "enforceForClassMembers": false }]</code>
+      </td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>
+        <code>["error", { "getWithoutSet": true }]</code>
+        <br />
+        <code>[2, { "getWithoutSet": true }]</code>
+      </td>
+      <td>No</td>
+    </tr>
+    <tr>
+      <td>
+        <code>["error", { "enforceForClassMembers": true }]</code>
+        <br />
+        <code>[2, { "enforceForClassMembers": true }]</code>
+      </td>
+      <td><strong>Yes</strong></td>
+    </tr>
+      <td>
+        <code>["error", { "getWithoutSet": false }]</code>
+        <br />
+        <code>[2, { "getWithoutSet": false }]</code>
+      </td>
+      <td><strong>Yes</strong></td>
+    </tr>
+  </tbody>
+</table>
+
+## Detailed Design
+
+Additional logic can be added to the existing code points in `Linter` that validate inline config options: [`_verifyWithFlatConfigArrayAndWithoutProcessors`](https://github.com/eslint/eslint/blob/7c78ad9d9f896354d557f24e2d37710cf79a27bf/lib/linter/linter.js#L1636).
+For a rough code reference, see [`poc: reporting unused inline configs`](https://github.com/JoshuaKGoldberg/eslint/commit/e14e404ed93e6238bdee817923a449f5215eecd8).
+
+### Computing Option Differences
+
+Each inline config comment will be compared against the existing configuration value(s) it attempts to override:
+
+- If the config comment only specifies a severity, then only the severity will be checked for redundancy
+  - The new logic will normalize options: `"off"` will be considered equivalent to `0`
+- If the config comment also specifies rule options, they will be compared for deep equality to the existing rule options
```

Comment:
If a config comment specifies a different severity but the same options as the file config, should there be a warning? I think the intended answer is no, but it's not obvious from the wording or from the other provided examples.

Here's [**an example**](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVudXNlZC12YXJzOiBbXCJvZmZcIiwgeyBcInZhcnNcIjogXCJhbGxcIiB9XSAqL1xuIiwib3B0aW9ucyI6eyJydWxlcyI6eyJuby11bnVzZWQtdmFycyI6WyJlcnJvciJdfSwibGFuZ3VhZ2VPcHRpb25zIjp7InNvdXJjZVR5cGUiOiJtb2R1bGUiLCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19) in the Playground.

Reply:
I would expect no; if the severity is different then the inline config is used.

Reply:
👍 that's the intent, yes. I'll make it clear.

Reply:
dbd33bc